test/e2e/framework/util.go

```diff
@@ -3265,19 +3261,25 @@ func NodeAddresses(nodelist *v1.NodeList, addrType v1.NodeAddressType) []string
 	return hosts
 }
 
-// NodeSSHHosts returns SSH-able host names for all schedulable nodes - this excludes master node.
-// It returns an error if it can't find an external IP for every node, though it still returns all
-// hosts that it found in that case.
+// NodeSSHHosts returns SSH-able host names for all schedulable nodes - this
+// excludes master node. If it can't find any external IPs, it falls back to
+// looking for internal IPs. If it can't find an internal IP for every node it
+// returns an error, though it still returns all hosts that it found in that
+// case.
 func NodeSSHHosts(c clientset.Interface) ([]string, error) {
 	nodelist := waitListSchedulableNodesOrDie(c)
 
-	// TODO(roberthbailey): Use the "preferred" address for the node, once such a thing is defined (#2462).
 	hosts := NodeAddresses(nodelist, v1.NodeExternalIP)
+	// If ExternalIP isn't set, assume the test programs can reach the InternalIP.
+	if len(hosts) == 0 {
+		log("INFO", "No external IP address on nodes, falling back to internal IPs")
```

Comment:
Added a log line to potentially clarify things if someone needs to debug issues around these tests. Will make it discernible from the logs which set of addresses were used.

Reply:
nit: use the `Logf` helper instead of calling `log` directly?